fish_prompt.fish

```diff
@@ -861,13 +861,31 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
 end
 
 function __bobthefish_prompt_nvm -S -d 'Display current node version through NVM'
-    [ "$theme_display_nvm" = 'yes' -a -n "$NVM_DIR" ]
-    or return
+    [ "$theme_display_nvm" = 'no' ]
+    and return
 
-    set -l node_version (nvm current 2> /dev/null)
+    set -l node_version
+    if type -fq nvm
```

Comment:
This could be a problem. Using my environment as an example:

```
 type -f nvm                                                              373ms
type: Could not find 'nvm'
```

but, since I've installed an OMF plugin...

```
 type nvm                                                                 185ms
nvm is a function with definition
# Defined in /Users/fuzzy/.local/share/omf/pkg/nvm/functions/nvm.fish @ line 1
function nvm --description 'Node version manager'
    if test -e $nvm_prefix/nvm.sh
        if not type -q fenv
            echo "You need to install foreign-env plugin"
            return 1
        end

        fenv source $nvm_prefix/nvm.sh --no-use ';' nvm $argv
    else
        echo "You need to install nvm itself (see https://github.com/creationix/nvm#installation)"
        return 1
    end
end
```

As you can see I do have `nvm` but the binary executable is wrapped and masked by a function. This is how a lot of plugins work AFAIK.

My supposition is that `type -f` check could not be safe enough. Any thoughts?

Reply:
Good point. In this case, `type -q` rather than `type -fq` seems appropriate.

Reply:
Thanks for pointing this out! I updated it to `type -q`. :+1: